### ISSUE TITLE
[Bug]: Clicking action button not selecting action button when action button is inside action group.

### ISSUE THREAD

**Issue by sonirajan** [adobe/spectrum-web-components#3048]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-action-group, sp-action-button

### Expected behavior

Clicking any action button should select it when they are child elements of action group.

### Actual behavior

I’m trying to use sp-action-group component with sp-action-button as child components. When I try to select/click one of the sp-action-button, it calls [handleClick](https://github.com/adobe/spectrum-web-components/blob/main/packages/action-group/src/ActionGroup.ts#L171) event. Here, event.target is always span instead of parent sp-action-button. And, because of that target.value is always "undefined" and returns and does nothing. Is this the bug on SWC side? I have sp-action-group v0.12.9 and sp-action-button v0.10.12 installed. Slack conversion is [here](https://creativecloud.slack.com/archives/CESK60MQD/p1678898210005339).

### What browsers are you seeing the problem in?

Chrome

**Comment by Westbrook** [comment]
Indeed. Seems like we could update
```
const target = event.target as ActionButton;
```
To something like:
```
const target = event.composedPath().find(el => this.buttons.includes(el));
```
But, performant and typescripty and supported by a test.

We'd appreciate a contribution on this!

**Comment by Westbrook** [comment]
fixed by #3292
